code/modules/chemistry/Reagents-PoisonEtc.dm

```diff
@@ -1713,14 +1713,12 @@ datum
 				if (!M) M = holder.my_atom
 				//M.changeStatus("radiation", 30, 1)
 				if (!src.data) // Pull bioholder data from blood that's in the same reagentholder
-					if(holder.has_reagent("bloodc"))
-						var/datum/reagent/blood/cheating = holder.reagent_list["bloodc"]
-						if (cheating && istype(cheating.data, /datum/bioHolder))
-							src.data = cheating.data
-					else if(holder.has_reagent("blood"))
-						var/datum/reagent/blood/cheating = holder.reagent_list["blood"]
-						if (cheating && istype(cheating.data, /datum/bioHolder))
-							src.data = cheating.data
+					for (var/bloodtype in list("blood", "bloodc", "hemolymph", "milk", "calcium"))
```

Comment:
Any reason not to just check every reagent in case we are future blood types? Performance concern?

Reply:
i doubt it would be a performance concern given that even charcoal has a for loop for all chems when purging them, just get the first one with valid DNA and then break the for loop

Reply:
good point!